subprojects/logging/src/main/java/org/gradle/internal/logging/progress/ProgressLoggerFactory.java

```diff
@@ -36,5 +38,16 @@
      */
     ProgressLogger newOperation(Class<?> loggerCategory);
 
+    /**
+     * Creates a new long-running operation which has not been started, associated
+     * with the given build operation id.
+     *
+     * @param loggerCategory The logger category.
+     * @param buildOperationId OperationIdentifier to be associated with all logs produced by
+     * the new ProgressLogger
+     * @return the progress logger for the operation.
+     */
+    ProgressLogger newOperation(Class<?> loggerCategory, OperationIdentifier buildOperationId);
+
     ProgressLogger newOperation(Class<?> loggerClass, ProgressLogger parent);
```

Comment:
Not yet, I don't think. `TaskExecutionLogger` is the only thing that does this, and I expect we'll change up how it does progress logging to build operations before too long.